pyiron_continuum/damask/damaskjob.py

```diff
@@ -1,157 +1,143 @@
             stress = np.zeros(len(stress_path))
-            hdf = h5py.File(d.fname)
+            hdf = h5py.File(self._damask_results.fname)
             for count,path in enumerate(stress_path):
                 stress[count] = np.array(hdf[path])
-            stress = np.array(stress)/1E6
-            return stress
+            self.output.stress = np.array(stress)/1E6
 
-    def eval_strain(self):
+    def strain(self):
         """
         return the strain as a numpy array
         Parameters
         ----------
         job_file : str
           Name of the job_file
         """
-        file_name = self.output_file
-        if file_name is not None:
-            d = Result(file_name)
-            stress_path = d.get_dataset_location('avg_sigma')
+        self.load_results()
+        if self._damask_results is not None:
+            stress_path = self._damask_results.get_dataset_location('avg_sigma')
             strain = np.zeros(len(stress_path))
-            hdf = h5py.File(d.fname)
+            hdf = h5py.File(self._damask_results.fname)
             for count,path in enumerate(stress_path):
-                strain[count] = np.array(hdf[path.split('avg_sigma')[0]     + 'avg_epsilon'])
+                strain[count] = np.array(hdf[path.split('avg_sigma')[0]+ 'avg_epsilon'])
+            self.output.strain = strain
 
-            return strain  
-        
-    def plot(self):
+    def to_hdf(self, hdf=None, group_name=None):
+        # TODO: check the function in the original function
+        self.input.to_hdf(hdf=self._hdf5, group_name='input')
+        self.output.to_hdf(hdf=self._hdf5, group_name='output')
+
+    @property
+    def plot_stress_strain(self):
         """
         Plot the stress strain curve from the job file
-
         Parameters
         ----------
-        job_file : str
-        Name of the job_file
+        self.stress
+        self.strain
         """
-        file_name = self.output_file
-        if file_name is not None:
-            d = Result(file_name)
-            stress_path = d.get_dataset_location('avg_sigma')
-            stress = np.zeros(len(stress_path))
-            strain = np.zeros(len(stress_path))
-            hdf = h5py.File(d.fname)
-            for count, path in enumerate(stress_path):
-                stress[count] = np.array(hdf[path])
-                strain[count] = np.array(hdf[path.split('avg_sigma')[0]     + 'avg_epsilon'])
-
-            stress = np.array(stress)/1E6
-            plt.plot(strain,stress,linestyle='-',linewidth='2.5')
-            plt.xlabel(r'$\varepsilon_{VM} $',fontsize=18)
-            plt.ylabel(r'$\sigma_{VM}$ (MPa)',fontsize=18)
\ No newline at end of file
+        #stress = self.stress()
+        #strain = self.strain()
+        plt.plot(self.output.strain, self.output.stress, linestyle='-', linewidth='2.5')
+        plt.xlabel(r'$\varepsilon_{VM} $', fontsize=18)
+        plt.ylabel(r'$\sigma_{VM}$ (MPa)', fontsize=18)
```

Comment:
Either we make this one a method again `job.plot_stress_strain()`, or we rename it to `stress_strain_plot`, i.e. making the `plot` a noun instead of a verb. Also right now, this property does not take any parameters 😅. 
 However, since this actually is a plotting method, I would like something like,

```
def plot_stress_strain(self, ax=None):
        """
        Plot the stress strain curve from the job file
        Parameters
        ----------
        ax (matplotlib axis /None): axis to plot on (created if None)
        """
        if ax is None:
            fig, ax = plt.subplots()
        ax.plot(self.output.strain, self.output.stress, linestyle='-', linewidth='2.5')
        ax.xlabel(r'$\varepsilon_{VM} $', fontsize=18)
        ax.ylabel(r'$\sigma_{VM}$ (MPa)', fontsize=18)

        return fig, ax
```